pkg/controller/rollout/batchrelease/inner_batchrelease.go

```diff
@@ -188,113 +184,47 @@ func (r *innerBatchRelease) Promote(index int32, isRollback, checkReady bool) (b
 	return false, nil
 }
 
-func (r *innerBatchRelease) resumeStableWorkload(checkReady bool) (bool, error) {
-	// cloneSet
-	switch r.rollout.Spec.ObjectRef.WorkloadRef.Kind {
-	case util.ControllerKruiseKindCS.Kind:
-		dName := r.rollout.Spec.ObjectRef.WorkloadRef.Name
-		obj := &appsv1alpha1.CloneSet{}
-		err := r.Get(context.TODO(), types.NamespacedName{Namespace: r.rollout.Namespace, Name: dName}, obj)
-		if err != nil {
-			if errors.IsNotFound(err) {
-				klog.Warningf("rollout(%s/%s) cloneSet(%s) not found, and return true", r.rollout.Namespace, r.rollout.Name, dName)
-				return true, nil
-			}
-			return false, err
-		}
-		// default partition.IntVal=0
-		if !obj.Spec.UpdateStrategy.Paused && obj.Spec.UpdateStrategy.Partition.IntVal == 0 && obj.Spec.UpdateStrategy.Partition.Type == intstr.Int {
+func (r *innerBatchRelease) resumeStableWorkload(waitReady bool) (bool, error) {
+	batch, err := r.FetchBatchRelease()
+	if err != nil {
+		if errors.IsNotFound(err) {
 			return true, nil
 		}
+		return false, err
+	}
 
-		err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-			if err = r.Get(context.TODO(), types.NamespacedName{Namespace: r.rollout.Namespace, Name: dName}, obj); err != nil {
-				return err
-			}
-			obj.Spec.UpdateStrategy.Paused = false
-			obj.Spec.UpdateStrategy.Partition = nil
-			return r.Update(context.TODO(), obj)
-		})
-		if err != nil {
-			klog.Errorf("update rollout(%s/%s) cloneSet failed: %s", r.rollout.Namespace, r.rollout.Name, err.Error())
-			return false, err
-		}
-		klog.Infof("resume rollout(%s/%s) cloneSet(paused=false,partition=nil) success", r.rollout.Namespace, r.rollout.Name)
-		return true, nil
-
-	case util.ControllerKindDep.Kind:
-		// deployment
-		dName := r.rollout.Spec.ObjectRef.WorkloadRef.Name
-		obj := &apps.Deployment{}
-		err := r.Get(context.TODO(), types.NamespacedName{Namespace: r.rollout.Namespace, Name: dName}, obj)
-		if err != nil {
-			if errors.IsNotFound(err) {
-				klog.Warningf("rollout(%s/%s) stable deployment(%s) not found, and return true", r.rollout.Namespace, r.rollout.Name, dName)
-				return true, nil
+	// If BatchPartition is nil, BatchRelease will directly resume workload via:
+	// - * set workload Paused = false if it needs;
+	// - * set workload Partition = null if it needs.
+	if batch.Spec.ReleasePlan.BatchPartition == nil {
+		// - If checkReady is true, finalizingPolicy must be "WaitResume";
+		// - If checkReady is false, finalizingPolicy must be NOT "WaitResume";
+		// otherwise, we should correct it.
+		switch batch.Spec.ReleasePlan.FinalizingPolicy {
```

Comment:
Is it possible to directly judge whether the status is completed, without judging waitReady?